### DISCUSSION TITLE
How a PVC related to the 1024 filesystems limitation?

### DISCUSSION THREAD

**Discussion by aledbf** [weka/csi-wekafs#171] [Q&A] [answered]
Does this mean I can only have 1024 PVCs?

**Comment by sergeyberezansky** [comment]
Hello,
Please take a look at the usage.md
There are 3 volume flavors in the new CSI 2.0
The filesystem-backed flavor is indeed limited by number of filesystems
Snapshot-backed volumes have much larger limits
Directory-backed (aka CSI 0.8.4) volumes are practically unlimited

**Comment by aledbf** [comment]
@sergeyberezansky thanks!

**Comment by aledbf** [comment]
@sergeyberezansky where can I find more details about the security model for the directory-backed volumes?
(I need to understand the isolation model between the volumes)

**Comment by sergeyberezansky** [comment, marked as answer]
1. Only the node that publishes any directory on a filesystem as a CSI volume, actually mounts that filesystem.
2. Then it performs a mount bind for the particular directory to a separate path, which in turn is bound inside the pod by Kubernetes. So the pod has a root of the volume set to be the directory
3. In new version, if different mount options are used via storage class definitions, node may mount the filesystem multiple times per actual set of mount options.
4. The node uses Weka authenticated mount to mount the filesystem (if supported by your Weka cluster version)
